### ISSUE TITLE
print a remote pdf

### ISSUE THREAD

**Issue by weihongyu12** [crabbly/Print.js#95]
Can you print a remote PDF? Existing Features Printing a remote file will report a cross-domain error, but in general, the file server may not be in the same domain as the application server

**Comment by crabbly** [comment]
@weihongyu12 , we can't print a remote pdf because of the cross-domain protection and the same-origin policy. 

I'll check if we there is way to work around it. Maybe we can implement a solution in the library.

**Comment by edsonrodsilva** [comment]
I'm also looking for a way to print a remote pdf.
Using this library we have cross-origin error frame.

Can you imagine a way to do it?

**Comment by weihongyu12** [comment]
Try [PDF.js](https://github.com/mozilla/pdf.js)?

**Comment by steveklebanoff** [comment]
You could create a server endpoint that acts a proxy.

i.e. `http://myapp.com/cool_pdf.pdf` could route to a server-side function that grabs the remote PDF and renders it.

**Comment by crabbly** [comment]
I haven't found a solid solution to work around the some-origin policy and allow the lib to print remove pdf files. As it uses iframe to load the pdf before printing it, the error will always occurs. 

I recommend using the solution above suggested by Steve.

**Comment by crabbly** [comment]
I implemented a better pdf pre-loading logic which I just realized it solved the issue when printing remote pdf files.
The file will be fetched via a http request and saved into a local blob url. When sending this url to the iframe, the browser won't block it since it is not longer a remote file.

Here is an example:
http://plnkr.co/edit/kxn1TuoP9Dxj75jrdl5W?p=preview

For now, it will only do this when preloading, which means, we have to pass the `showModal` parameter as `true` or use a hook function `onLoadingStart`. 
I'm not sure that is worth to implement the same method by default (without preload / loading hooks). 

Either way, I thought this was pretty cool and wanted to share.
Run `npm update print-js` to get the latest version.
